src/git-repository-async.js

```diff
@@ -0,0 +1,255 @@
+'use babel'
+
+const Git = require('nodegit')
+const path = require('path')
+const {Emitter, Disposable, CompositeDisposable} = require('event-kit')
+
+// Temporary requires
+// ==================
+// GitUtils is temporarily used for ::relativize only, because I don't want
+// to port it just yet. TODO: remove
+const GitUtils = require('git-utils')
+// Just using this for _.isEqual and _.object, we should impl our own here
+const _ = require('underscore-plus')
+
+module.exports = class GitRepositoryAsync {
+  static open (path, options = {}) {
+    // QUESTION: Should this wrap Git.Repository and reject with a nicer message?
+    return new GitRepositoryAsync(path, options)
+  }
+
+  constructor (path, options) {
+    this.repo = null
+    this.emitter = new Emitter()
+    this.subscriptions = new CompositeDisposable()
+    this.pathStatusCache = {}
+    this._gitUtilsRepo = GitUtils.open(path) // TODO remove after porting ::relativize
+    this.repoPromise = Git.Repository.open(path)
+
+    var {project, refreshOnWindowFocus} = options
+    this.project = project
+    if (refreshOnWindowFocus === undefined) {
```

Comment:
I actually want to check undefined here, because it should default to `true` - that is to say, `true` is `true` and `undefined` is `true`, but only explicitly set`false` is `false`, which isn't quite how the Boolean constructor works.